docs/reference/search/search-your-data/semantic-search-semantic-text.asciidoc

```diff
@@ -0,0 +1,267 @@
+<3> The `inference_id` is the inference endpoint you created in the previous step.
+It will be used to generate the tokens based on the input text.
+Every time you ingest data into the related `semantic_text` field, this endpoint will be used for creating the vector representation of the text.
+<4> The field to store the text reindexed from a source index in the  <<semantic-text-reindex-data,Reindex the data>> step.
+<5> The textual data stored in the `content` field will be copied to `semantic_text` and processed by the {infer} endpoint.
+The `semantic_text` field will store the embeddings generated based on the input data.
+
+
+[discrete]
+[[semantic-text-load-data]]
+==== Load data
+
+In this step, you load the data that you later use to create embeddings from it.
+
+Use the `msmarco-passagetest2019-top1000` data set, which is a subset of the MS
+MARCO Passage Ranking data set. It consists of 200 queries, each accompanied by
+a list of relevant text passages. All unique passages, along with their IDs,
+have been extracted from that data set and compiled into a
+https://github.com/elastic/stack-docs/blob/main/docs/en/stack/ml/nlp/data/msmarco-passagetest2019-unique.tsv[tsv file].
+
+Download the file and upload it to your cluster using the
+{kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[Data Visualizer]
+in the {ml-app} UI. Assign the name `id` to the first column and `content` to
+the second column. The index name is `test-data`. Once the upload is complete,
+you can see an index named `test-data` with 182469 documents.
+
+
+[discrete]
+[[semantic-text-reindex-data]]
+==== Reindex the data
+
+Create the embeddings from the text by reindexing the data from the `test-data`
+index to the `semantic-embeddings` index. The data in the `content` field will
+be reindexed into the `content` field of the destination index. Then the same
+data will be copied to the `semantic_text` field as a result of the `copy_to`
+parameter set in the index mapping creation step. The copied data will be
+processed by the {infer} task associated with the `semantic_text` semantic text
+field.
+
+[source,console]
+------------------------------------------------------------
+POST _reindex?wait_for_completion=false
+{
+  "source": { 
+    "index": "test-data",
+    "size": 10 <1>
+  },
+  "dest": {
+    "index": "semantic-embeddings"
+  }
+}
+------------------------------------------------------------
+// TEST[skip:TBD]
+<1> The default batch size for reindexing is 1000. Reducing size to a smaller
+number makes the update of the reindexing process quicker which enables you to
+follow the progress closely and detect errors early.
+
+The call returns a task ID to monitor the progress:
+
+[source,console]
+------------------------------------------------------------
+GET _tasks/<task_id>
+------------------------------------------------------------
+// TEST[skip:TBD]
+
+It is recommended to cancel the reindexing process if you don't want to wait
```

Comment:
Maybe start with a smaller dataset for the example so it's guaranteed to be fast?

Reply:
All semantic search-related tutorials use the same dataset. I plan to find something smaller (around 2000-5000 data points) and amend all the tutorials to use that smaller set. However, it's an effort of an upcoming sprint, so I'd rather go ahead with this content and adjust all the docs later in one go.